src/pybind/mgr/orchestrator/module.py

```diff
@@ -1083,6 +1083,13 @@ def _apply_rgw(self,
         if inbuf:
             raise OrchestratorValidationError('unrecognized command -i; -h or --help for usage')
 
+        if realm and not zone:
```

Comment:
and what happens if the user does not specify zone and does not specify realm?
If we always need to set realm and zone  better set both parameters mandatory and not optional

Reply:
not setting both is a valid option. that would deploy default(nonmultisite) mode